### PR TITLE
genconf command added

### PR DESCRIPTION
**Pull request by patriki01**

### CODE REVIEW

File: sssd_test_framework/utils/sssd.py [hunk 1]
```diff
@@ -389,6 +389,26 @@ def config_apply(self, check_config: bool = True, debug_level: str | None = "0xf
...
+    def genconf(self) -> SSHProcessResult:
+        """
+        Exec ``sssd --genconf``.
+
+        :return: Result of ran command.
+        :rtype: SSHProcessResult
+        """
+        return self.host.ssh.exec(["/usr/sbin/sssd", "--genconf"])
+
+    def genconf_section(self, section: str) -> SSHProcessResult:
```
Comment (pbrezina) [inline sssd_test_framework/utils/sssd.py:401]: This can be merged into genconf method.

**Review by pbrezina — CHANGES_REQUESTED** [review]

File: sssd_test_framework/utils/sssd.py (hunk 1 above)
Comment (patriki01) [reply sssd_test_framework/utils/sssd.py]: Done

File: sssd_test_framework/utils/sssd.py [hunk 2]
```diff
@@ -389,6 +389,22 @@ def config_apply(self, check_config: bool = True, debug_level: str | None = "0xf
         if check_config:
             self.host.ssh.run("sssctl config-check")
 
+    def genconf(self, section: str | None = None) -> SSHProcessResult:
+        """
+        Exec ``sssd --genconf`` or ``sssd --genconf-section=section`` while section is not None.
+
+        :param section: Section that will be refreshed. Defaults to None.
```
Comment (pbrezina) [inline sssd_test_framework/utils/sssd.py:396]: \``None``

File: sssd_test_framework/utils/sssd.py [hunk 3]
```diff
@@ -389,6 +389,22 @@ def config_apply(self, check_config: bool = True, debug_level: str | None = "0xf
         if check_config:
             self.host.ssh.run("sssctl config-check")
 
+    def genconf(self, section: str | None = None) -> SSHProcessResult:
+        """
+        Exec ``sssd --genconf`` or ``sssd --genconf-section=section`` while section is not None.
```
Comment (pbrezina) [inline sssd_test_framework/utils/sssd.py:394]: if \``section`` is not \``None``.

**Review by pbrezina — CHANGES_REQUESTED** [review]
Please change the commit message to "sssd: add genconf method"

**Review by pbrezina — CHANGES_REQUESTED** [review]

File: sssd_test_framework/utils/sssd.py [hunk 4]
```diff
@@ -389,6 +389,22 @@ def config_apply(self, check_config: bool = True, debug_level: str | None = "0xf
...
+        :type path: str | None, optional
+        :return: Result of the ran command.
+        :rtype: SSHProcessResult
+        """
+        if section is None:
+            cmd = self.host.ssh.exec(["/usr/sbin/sssd", "--genconf"])
+        else:
+            cmd = self.host.ssh.exec(["/usr/sbin/sssd", f"--genconf-section={section}"])
+
+        return cmd
```
[the remark this answers is not in the document]
Comment (patriki01) [reply sssd_test_framework/utils/sssd.py]: I thought that the way I did it it was more clear. But I see now it is not. FIXED.

**Review by pbrezina — APPROVED** [review]

**Pull request merged**